src/HttpMethodsClient.php

```diff
@@ -11,6 +11,14 @@
  *
  * Use this interface when you do not have PSR-7 RequestInterface instances available.
  *
+ * Implementations should be configured in their constructor. For runtime behaviour changes, there
+ * is an $options parameter. It SHOULD be validated by the client and an exception thrown if
+ * unknown options are passed.
```

Comment:
> As an exemple, for our library (docker-php) we need to have this
> settings per request, as i want to have a logs response in a async
> manner (it's a chunked reponse which is populated from consecutive
> actions and stay alive "forever"), but we also want to have the stop /
> start calls in a sync manner (to be sure that it is correctly stop /
> start before doing anything else).

i think this is exactly an example where you could just use two client
instances: a synchronous one for the start/stop and an asynchronous for
the logging. that way is a lot more inversion of control. your library
specifies that it needs a sync client and one that is efficient for
logging. it does not need to know the options to make that happen. the
application bootstrap just creates the two clients and gives them to
your library.

i think i am for entirely dropping the $options argument. it is just an
invitation to write client specific code, and the purpose of this
abstraction is to avoid that.